indexer/altitude_loader.cpp

```diff
@@ -0,0 +1,110 @@
+#include "indexer/altitude_loader.hpp"
+
+#include "coding/reader.hpp"
+
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/thread.hpp"
+
+#include "defines.hpp"
+
+#include "3party/succinct/mapper.hpp"
+
+namespace
+{
+void ReadBuffer(ReaderSource<FilesContainerR::TReader> & rs, vector<char> & buf)
+{
+  uint32_t bufSz = 0;
+  rs.Read(&bufSz, sizeof(bufSz));
+  if (bufSz > rs.Size() + rs.Pos())
+  {
+    ASSERT(false, ());
+    return;
+  }
+  buf.clear();
+  buf.resize(bufSz);
+  rs.Read(buf.data(), bufSz);
+}
+} // namespace
+
+namespace feature
+{
+AltitudeLoader::AltitudeLoader(MwmValue const * mwmValue)
+  : reader(mwmValue->m_cont.GetReader(ALTITUDE_FILE_TAG)), m_altitudeInfoOffset(0), m_minAltitude(kInvalidAltitude)
+{
+  if (!mwmValue || mwmValue->GetHeader().GetFormat() < version::Format::v8 )
+    return;
+
+  try
+  {
+    ReaderSource<FilesContainerR::TReader> rs(reader);
+    DeserializeHeader(rs);
+
+    // Reading rs_bit_vector with altitude availability information.
+    ReadBuffer(rs, m_altitudeAvailabilitBuf);
+    m_altitudeAvailability = make_unique<succinct::rs_bit_vector>();
```

Comment:
On the other hand, while I redesigned Value constructor (now it takes only one param) I has to wrap in unique_ptr altitudeLoader member. So it's possible to get read of unique_ptr around succincts. Done.